### PR TITLE
add capability of estimating rain number concentration from rain mixing ratio

### PR DESCRIPTION
**Pull request by daviddowellNOAA**
Analyzing rain number concentration with EnKF / EnVar radar-reflectivity data assimilation remains a research subject.  But we do get useful rain mixing ratios from the radar DA, and we need to pair the mixing ratio estimates with reasonable values of number concentration.  The cloud analysis runs immediately after radar DA and before the forecast and is thus the appropriate place to initialize rain number concentration.  The number concentration is computed from mixing ratio by assuming the drop sizes have an exponential distribution, with the intercept parameter specified with a new namelist parameter "n0_rain".  This new capability is turned on during the cloud analysis by setting the new namelist parameter "l_qnr_from_qr" to true.  The default value of "l_qnr_from_qr" is false.

### CODE REVIEW

**Review by terraladwig — APPROVED** [review]
These changes look good to me.

**Comment by christopherwharrop-noaa** [comment]
Were any tests run to verify that turning on the new option yields the intended behavior?  This code is super simple and I certainly wouldn't expect any issues, but running a test is always more reassuring than relying on eyeballing the Fortran code.

**Comment by daviddowellNOAA** [comment]
Thanks for the question.  I ran two tests.  First, I confirmed no change to the output of the cloud analysis when l_qnr_from_qr is .false.  After a second test, with l_qnr_from_qr set to .true., I made plots of the resulting rain number concentration.  The results look realistic.  I think it's safe to try this new capability in the hourly-cycled ensemble.

**Comment by christopherwharrop-noaa** [comment]
Fantastic!  Exactly what I was looking for.  We may add a PR template in the near future with a `TESTS` section where this information can be added in the description.  Thank you for doing those tests and for describing them here.

**Comment by christopherwharrop-noaa** [comment]
@hu5970 - I believe this PR is waiting on a review from you.  If you are around, could you please take a look?

**Review by hu5970 — APPROVED** [review]

**Pull request merged**
